internal/controller/atlasnetworkcontainer/atlasnetworkcontainer_controller.go

```diff
@@ -0,0 +1,136 @@
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/controller/atlas"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/controller/customresource"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/controller/reconciler"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/indexer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+)
+
+// AtlasNetworkContainerReconciler reconciles a AtlasNetworkContainer object
+type AtlasNetworkContainerReconciler struct {
+	reconciler.AtlasReconciler
+	AtlasProvider            atlas.Provider
+	Scheme                   *runtime.Scheme
+	EventRecorder            record.EventRecorder
+	GlobalPredicates         []predicate.Predicate
+	ObjectDeletionProtection bool
+	independentSyncPeriod    time.Duration
+}
+
+// type reconcileRequest struct {
+// 	workflowCtx      *workflow.Context
+// 	service          networkcontainer.NetworkContainerService
+// 	projectID        string
+// 	networkContainer *akov2.AtlasNetworkContainer
+// }
+
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=atlasnetworkcontainers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=atlasnetworkcontainers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=atlasnetworkcontainers/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the AtlasNetworkContainer object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.4/pkg/reconcile
+func (r *AtlasNetworkContainerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Infow("-> Starting AtlasNetworkContainer reconciliation")
+
+	networkContainer := akov2.AtlasNetworkContainer{}
+	result := customresource.PrepareResource(ctx, r.Client, req, &networkContainer, r.Log)
+	if !result.IsOk() {
+		return result.ReconcileResult(), errors.New(result.GetMessage())
+	}
+	return r.handleCustomResource(ctx, &networkContainer)
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *AtlasNetworkContainerReconciler) SetupWithManager(mgr ctrl.Manager, skipNameValidation bool) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&akov2.AtlasNetworkContainer{}).
```

Comment:
**blocking**: this is missing Global Predicates
```suggestion
		For(&akov2.AtlasNetworkContainer{}, builder.WithPredicates(r.GlobalPredicates...)).
```

Reply:
also a big ask, could you please change this to:
```suggestion
		For(r.For()).
```
where
```
func (r *AtlasNetworkContainerReconciler) For() (client.Object, builder.Predicates) {
	return &akov2.AtlasNetworkContainer{}, builder.WithPredicates(r.GlobalPredicates...)
}
```

to make it compatible with https://github.com/mongodb/mongodb-atlas-kubernetes/pull/2080

Reply:
is `For()` already merged? I can wait if not

Reply:
ah this is just an interface, I can already implement. Good! 👌

Reply:
Done